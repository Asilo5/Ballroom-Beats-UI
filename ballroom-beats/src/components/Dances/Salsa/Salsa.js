import React, { Component } from 'react';
import {
    View,
    Animated,
    Easing,
    TouchableOpacity,
    StyleSheet
} from 'react-native';

export default class Salsa extends Component {
  
    state = {
        pulses: [
            new Animated.Value(1),
            new Animated.Value(1),
            new Animated.Value(1),
            new Animated.Value(1),
            new Animated.Value(1),
            new Animated.Value(1)
        ],
        counter: 0
    };

    componentDidMount() {
        this.props.start(this.generateTiming());
        this.props.stopMusic(this.getSongLength());
      };
    
      endGame = (userPoints, gamePoints) => {
          this.props.stopDance(userPoints, gamePoints);
      };
    
      componentWillUnmount() {
        this.endGame(this.state.counter, this.getExpectedValue());
      };
    
      getExpectedValue = () => {
        return this.props.song.duration/60 * this.props.song.tempo;
      };
    
      getSongLength = () => {
        //Real World
    
        // return (
        //   this.props.song.duration * 1000
        // )
    
        //Demo
        return 25000;
      };
    
      assessPulseDuration = () => {
        return (
          49500/this.props.song.tempo
        )
      };
    
      generateTiming = () => {
        return this.state.pulses.map(pulse => {
          return [
            Animated.timing(pulse, {
              toValue: 6,
              duration: this.assessPulseDuration(),
              easing: Easing.back(),
            }),
            Animated.timing(pulse, {
              toValue: 1,
              duration: this.assessPulseDuration(),
            })
          ]
        }).flat();
      };


   countUp = () => {
       let newCount = this.state.counter;
       newCount++;
       this.setState({ counter: newCount });
   } 

   generateViews = () => {
       return (
           <View style={styles.danceFloor}>
               <TouchableOpacity style={styles.loneDot} onPress={() => this.countUp()} key={0}>
                   <Animated.View
                       style={{
                        transform: [
                            {
                                scaleX: this.state.pulses[0]
                            },
                            {
                               scaleY: this.state.pulses[0]
                           }
                        ],
                        margin: 20,
                        borderWidth: 15,
                        borderColor: "#FFEB00",
                        borderRadius: 15
                    }}
                   >

                   </Animated.View>
               </TouchableOpacity>
               
               <View style={styles.middleSteps}>
                   <TouchableOpacity onPress={() => this.countUp()} key={1}>
                      <Animated.View
                        style={{
                          transform: [
                              {
                                  scaleX: this.state.pulses[1]
                              },
                              {
                                 scaleY: this.state.pulses[1]
                             }
                          ],
                          margin: 20,
                          borderWidth: 15,
                          borderColor: "#F6811F",
                          borderRadius: 15
                      }}
                      >
                       
                       </Animated.View>
                   </TouchableOpacity>

                   <TouchableOpacity onPress={() => this.countUp()} key={2}>
                      <Animated.View
                        style={{
                          transform: [
                              {
                                  scaleX: this.state.pulses[2]
                              },
                              {
                                 scaleY: this.state.pulses[2]
                             }
                          ],
                          margin: 20,
                          borderWidth: 15,
                          borderColor: "#03ABF0",
                          borderRadius: 15
                      }}
                      >
                       
                       </Animated.View>
                   </TouchableOpacity>
               </View>

               <View style={styles.middleSteps}>
                   <TouchableOpacity onPress={() => this.countUp()} key={5}>
                      <Animated.View
                        style={{
                          transform: [
                              {
                                  scaleX: this.state.pulses[5]
                              },
                              {
                                 scaleY: this.state.pulses[5]
                             }
                          ],
                          margin: 20,
                          borderWidth: 15,
                          borderColor: "#F6811F",
                          borderRadius: 15
                      }}
                      >
                       
                       </Animated.View>
                   </TouchableOpacity>

                   <TouchableOpacity onPress={() => this.countUp()} key={4}>
                      <Animated.View
                        style={{
                          transform: [
                              {
                                  scaleX: this.state.pulses[4]
                              },
                              {
                                 scaleY: this.state.pulses[4]
                             }
                          ],
                          margin: 20,
                          borderWidth: 15,
                          borderColor: "#03ABF0",
                          borderRadius: 15
                      }}
                      >
                       
                       </Animated.View>
                   </TouchableOpacity>
               </View>

               <TouchableOpacity style={styles.loneDot} onPress={() => this.countUp()} key={3}>
                  <Animated.View
                    style={{
                      transform: [
                          {
                              scaleX: this.state.pulses[3]
                          },
                          {
                             scaleY: this.state.pulses[3]
                         }
                      ],
                      margin: 20,
                      borderWidth: 15,
                      borderColor: "#6F2C8F",
                      borderRadius: 15
                  }}
                  >
                       
                  </Animated.View>
               </TouchableOpacity>
           </View>
       )
   }


   render() {
    return (
      <View>
        <View style={styles.stepsContainer}>
            {this.generateViews()}
        </View>
      </View>
    );
  };


};

const styles = StyleSheet.create({

  danceFloor: {
      display: "flex",
      justifyContent: "space-around",
      top: 200,
      padding: 50,
      left: 10
    },

    middleSteps: {
      display: "flex",
      flexDirection: "row",
      justifyContent: "space-around",
      padding: 20
    },

    stepsContainer: {
      marginBottom: 60,
      width: 600
    },

    loneDot: {
      width: '5%',
      left: '40%',
      padding: 20
    }

});