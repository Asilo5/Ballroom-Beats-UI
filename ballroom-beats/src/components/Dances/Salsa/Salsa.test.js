import React from 'react';
import renderer from 'react-test-renderer';
import Salsa from './Salsa';


describe('Salsa', () => {

   it('should match snapshot', () => {
    const salsa = renderer.create(<Salsa
      start={jest.fn()}
      song={jest.fn()}
      stopDance={jest.fn()}
      stopMusic={jest.fn()}
      />).toJSON();
    expect(salsa).toMatchSnapshot();
   });

});
