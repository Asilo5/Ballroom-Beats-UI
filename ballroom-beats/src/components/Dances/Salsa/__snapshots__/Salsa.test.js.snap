// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Salsa should match snapshot 1`] = `
<View>
  <View
    style={
      Object {
        "marginBottom": 60,
        "width": 600,
      }
    }
  >
    <View
      style={
        Object {
          "display": "flex",
          "justifyContent": "space-around",
          "left": 10,
          "padding": 50,
          "top": 100,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 20,
            "left": 200,
            "marginBottom": -25,
            "marginTop": -25,
          }
        }
      >
        Left Front
      </Text>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "left": "35%",
            "opacity": 1,
            "padding": 20,
            "width": "5%",
          }
        }
      >
        <View
          style={
            Object {
              "borderColor": "#F60091",
              "borderRadius": 24,
              "borderWidth": 24,
              "margin": 20,
              "transform": Array [
                Object {
                  "scaleX": 1,
                },
                Object {
                  "scaleY": 1,
                },
              ],
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "space-around",
            "padding": 20,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "borderColor": "#FFEB00",
                "borderRadius": 24,
                "borderWidth": 24,
                "margin": 20,
                "transform": Array [
                  Object {
                    "scaleX": 1,
                  },
                  Object {
                    "scaleY": 1,
                  },
                ],
              }
            }
          />
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 20,
                "left": 24,
              }
            }
          >
            Right
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "borderColor": "#F6811F",
                "borderRadius": 24,
                "borderWidth": 24,
                "margin": 20,
                "transform": Array [
                  Object {
                    "scaleX": 1,
                  },
                  Object {
                    "scaleY": 1,
                  },
                ],
              }
            }
          />
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 20,
                "left": 24,
              }
            }
          >
            Left
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "space-around",
            "padding": 20,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "borderColor": "#71C043",
                "borderRadius": 24,
                "borderWidth": 24,
                "margin": 20,
                "transform": Array [
                  Object {
                    "scaleX": 1,
                  },
                  Object {
                    "scaleY": 1,
                  },
                ],
              }
            }
          />
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 20,
                "left": 24,
              }
            }
          >
            Right
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "borderColor": "#03ABF0",
                "borderRadius": 24,
                "borderWidth": 24,
                "margin": 20,
                "transform": Array [
                  Object {
                    "scaleX": 1,
                  },
                  Object {
                    "scaleY": 1,
                  },
                ],
              }
            }
          />
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 20,
                "left": 24,
              }
            }
          >
            Left
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "left": "35%",
            "opacity": 1,
            "padding": 20,
            "width": "5%",
          }
        }
      >
        <View
          style={
            Object {
              "borderColor": "#6F2C8F",
              "borderRadius": 24,
              "borderWidth": 24,
              "margin": 20,
              "transform": Array [
                Object {
                  "scaleX": 1,
                },
                Object {
                  "scaleY": 1,
                },
              ],
            }
          }
        />
      </View>
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 20,
            "left": 200,
            "marginBottom": -25,
            "marginTop": -25,
          }
        }
      >
        Right Back
      </Text>
    </View>
  </View>
  <View />
</View>
`;
